tests/models/userModelTest.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+const User = require('../../models/userModel');
+const chai = require('chai');
+const mongoose = require('mongoose');
+const validator = require("validator");
+const expect = chai.expect;
+chai.use(require('chai-as-promised'));
+const dotenv = require('dotenv');
+dotenv.config({
+    path: './config.env'
+});
+
+before(async function () {
+    const database = process.env.DATABASE.replace(
+        '${MONGO_USERNAME}', process.env.MONGO_USERNAME).replace(
+        '${MONGO_PASSWORD}', process.env.MONGO_PASSWORD).replace(
+        '${MONGO_HOSTNAME}', process.env.MONGO_HOSTNAME).replace(
+        '${MONGO_PORT}', process.env.MONGO_PORT).replace(
+        '${MONGO_DB}', process.env.MONGO_DB_TEST);
+
+    console.log(database);
+
+// Connect the database
+    await mongoose.connect(database, {
+        useNewUrlParser: true,
+        useCreateIndex: true,
+        useFindAndModify: false,
+        useUnifiedTopology: true
+    }).then(con => {
+        console.log('DB connection Successfully!');
+        mongoose.connection.db.dropDatabase();
+    });
+
+});
+
+describe('UserModel', function () {
+    beforeEach(function (done) {
+        mongoose.connection.db.dropDatabase(() => {
+            done();
+        });
+    });
+    describe('Missing Element', function () {
+        Object.keys(User.schema.obj).forEach((key) => {
+            if (User.schema.obj[key].hasOwnProperty('required')) {
+                let test = {
+                    firstName: "test",
+                    lastName: "test",
+                    email: "email@email.test",
+                    password: "012345678",
+                };
+                delete test[key];
+                it('should throws when ' + key + ' is not present', async () => {
+                    await expect(User.create(test)).to.be.rejectedWith(Error);
+                });
+            }
+        });
+    });
+    describe('Role', function () {
+        it('should throws when value is not in the enum', async () => {
+            await expect(User.create({
+                firstName: "test",
+                lastName: "test",
+                email: "email@email.test",
+                password: "012345678",
+                role: "NOPE"
+            })).to.be.rejectedWith(Error);
+        });
+    });
+    describe('Role', function () {
```

Comment:
duplicate with line 58

Reply:
Yes. Was a bit to liberal with my copy-paste